### PR TITLE
Add caching for get struct Fields

### PR DESCRIPTION
**Pull request by kasugamirai**

### CODE REVIEW

File: pkl/decode_struct.go [hunk 1]
```diff
@@ -299,7 +299,12 @@ func parseStructOpts(field *reflect.StructField) structFieldOpts {
 	return ret
 }
 
+var structFieldsCache = make(map[reflect.Type]map[string]structField)
```
Comment (bioball) [inline pkl/decode_struct.go:302]: This isn't thread-safe, and also allocates structs that never get cleaned up, which isn't ideal.

What's the motivation behind this change? I'd be open to this change if we had some data that shows that this provides a performance improvement.

Comment (kasugamirai) [reply pkl/decode_struct.go:302]: Sorry, my mistake. I didn’t consider the thread safety and memory cleanup aspects. Thank you for pointing that out
